governance.md

```diff
@@ -0,0 +1,340 @@
+they disagree with the overall direction the BDFL is taking. The BDFL is chair
+of the Steering Council (see below) and may delegate their authority on a
+particular decision or set of decisions to any other Council member at their
+discretion.
+
+The BDFL can appoint their successor, but it is expected that the Steering
+Council would be consulted on this decision. If the BDFL is unable to appoint a
+successor, the Steering Council will make a suggestion or suggestions to the
+Main NumFOCUS Board. While the Steering Council and Main NumFOCUS Board will
+work together closely on the BDFL selection process, the Main NUMFOCUS Board
+will make the final decision.
+
+
+Steering Council
+----------------
+
+The Project will have a Steering Council that consists of Project Contributors
+who have produced contributions that are substantial in quality and quantity,
+and sustained over at least one year. The overall role of the Council is to
+ensure, through working with the BDFL and taking input from the Community, the
+long-term well-being of the project, both technically and as a community.
+
+During the everyday project activities, council members participate in all
+discussions, code review and other project activities as peers with all other
+Contributors and the Community. In these everyday activities, Council Members
+do not have any special power or privilege through their membership on the
+Council. However, it is expected that because of the quality and quantity of
+their contributions and their expert knowledge of the Project Software and
+Services that Council Members will provide useful guidance, both technical and
+in terms of project direction, to potentially less experienced contributors.
+
+The Steering Council and its Members play a special role in certain situations.
+In particular, the Council may:
+
+-   Make decisions about the overall scope, vision and direction of the
+    project.
+-   Make decisions about strategic collaborations with other organizations or
+    individuals.
+-   Make decisions about specific technical issues, features, bugs and pull
+    requests. They are the primary mechanism of guiding the code review process
+    and merging pull requests.
+-   Make decisions about the Services that are run by The Project and manage
+    those Services for the benefit of the Project and Community.
+-   Make decisions when regular community discussion doesn’t produce consensus
+    on an issue in a reasonable time frame.
+
+### Council membership
+
+To become eligible for being a Steering Council Member an individual should
+
+-  Be a Project Contributor who has produced contributions that are substantial in
+quality and quantity
+-  Sustain this development consistently over several months
+-  Support the community through code review, answering user questions,
+   triaging bug reports, participating constructively in broader conversations,
+   and being willing to help with documentation and infrastructure.
```

Comment:
Would it be reasonable to simplify this to be an engaged and active community member? Asking as there may be good candidates that focus on specific areas that we would still want to recruit them to fill more of a leadership role (assuming they are interested of course 😉).

Reply:
What does it mean to be an engaged and active community member?

In my mind there is likely to be differing thoughts on what this means.  My intention behind this item is to make it clear that people need to be engaged in ways other than contributing code to solve their problems.  Having direction over the project as a whole requires understanding several code facets of the project (arrays, dataframes, scheduling), several facets of maintenance (code review, user issue handling, release management), and a history of service (solving other people's problems).

People may consider themselves engaged because they use Dask at work, and continuously push functionality that they need into the project.  In my mind this makes them an excellent contributor, but not yet eligible to be on a decision-making council.

However, as I re-read these words I recognize that it sounds like a council member must do all of these things and so, someone who never wrote documentation would be ineligible.  That was not my intent and I'd be happy to change things around.

Thoughts?

Reply:
That was generally what I was wondering when reading the governance text previously.

My impression is we will be able to recruit people that are strong in a few of these areas, but not all. So I think we would benefit by acknowledging we welcome a diversity of skillsets into council membership. Some simplification to that effect might be helpful.

Reply:
Note that the  since introducing this says that just some of the areas should be not, not all.

Reply:
I've pushed a commit that turns this into a list of qualities that are desirable while generally highlighting this individual needs to be involved in various activities. Have also hinted this list might not be comprehensive. Please take a look and let me know what you think. 🙂